### PR TITLE
[WebTransport] Support latest HTTP datagram

### PR DESCRIPTION
**Pull request by yutakahirano**
Until the protocol is formally implemented by aioquic, we need to talk to
clients talking with the protocol. Let's add some ad-hoc support.

### CODE REVIEW

**Comment by yutakahirano** [comment]
@nidhijaju @vasilvv can you take a look?

File: tools/webtransport/h3/webtransport_h3_server.py [hunk 1]
```diff
@@ -72,6 +72,38 @@ def encode(self) -> bytes:
         return buffer.data
 
 
+class MyH3Connection(H3Connection):
```
Comment (vasilvv) [inline tools/webtransport/h3/webtransport_h3_server.py:75]: We should probably have a more descriptive name for this, unless the "My" prefix is a convention.

**Review by vasilvv — APPROVED** [review]
Overall LGTM. Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?

**Review by nidhijaju — APPROVED** [review]
I wasn't able to test that this works in my local environment as I still only see the "Opening handshake failed."/unresponsive page error when running the datagrams WPTs, but logic-wise looks okay to me.

File: tools/webtransport/h3/webtransport_h3_server.py (hunk 1 above)
Comment (yutakahirano) [reply tools/webtransport/h3/webtransport_h3_server.py]: Done.

**Comment by yutakahirano** [comment]
> Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?

I'll add the check to #30982.

**Comment by yutakahirano** [comment]
>> Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?
>
> I'll add the check to #30982.

I changed my mind, this is now addressed.

**Comment by yutakahirano** [comment]
Hi @web-platform-tests/wpt-core-team, I'd like to ask for your help. This change is blocked on some flaky tests (see [this](https://github.com/web-platform-tests/wpt/pull/31049/checks?check_run_id=3768302024) for example), but that's weird. We haven't run WebTransport server on CI bots so the WPT server is running without WebTransport over HTTP/3 support, and in that experiment I'm expect the tests (e.g.,. `/webtransport/datagrams.https.any.serviceworker.html`) fail consistently, which matches with my local results. 

Thanks!

**Review by bashi — APPROVED** [review]

**Comment by yutakahirano** [comment]
``

**Comment by yutakahirano** [comment]
Disabling (some) webtransport/ tests with .ini files.

**Pull request merged**